### DISCUSSION TITLE
Resetting a single field on a form with setValue - dirty status doesn't change

### DISCUSSION THREAD

**Discussion by MoSheikh** [react-hook-form/react-hook-form#5902] [Q&A] [answered]
Hi, 

I'm trying to implement a system where I can reset a single input field to the default value. Although I can use ```setValue``` to do this, after examining the ```dirtyFields``` or ```isDirty``` properties of the form state, even though the values are returned to their default values, the ```isDirty``` check does not occur, so the field is incorrectly marked as dirty. Simply or removing a character and changing it back to the reset value will properly trigger the ```isDirty``` check. 

I'm wondering is there anything in the API that I missed that would allow me to retrigger the ```isDirty``` check for a single field or for all fields? I do not wish to reset all the fields using the ```reset``` function, but just to recheck the input field values against the default values to ensure the  ```isDirty``` status is up to date.

I've modified an example I found to demonstrate this. 

### Functioning correctly with keystrokes:
1. Change the value in the text input using a keystroke (e.g. add a character or delete a character)
2. Press the ```Log``` button - the ```dirtyFields``` will have the ```a``` field marked as dirty
3. Change the value to the original default using keyboard inputs (e.g. delete the added character or add back the deleted character)
4. Press the ```Log``` button - the ```dirtyFields``` will be empty

### Functioning incorrectly with ```setValue```:
1. Change the value in the text input using a keystroke (e.g. add a character or delete a character)
2. Press the ```Log``` button - the ```dirtyFields``` will have the ```a``` field marked as dirty
3. Change the value to the original default using the ```Reset``` button which calls ```setValue```
4. Press the ```Log``` button - the ```dirtyFields``` will still show ```a``` marked as dirty

https://codesandbox.io/s/unruffled-lumiere-ly6q4?file=/src/App.tsx

Any help with this would be greatly appreciated.

**Comment by bluebill1049** [comment, marked as answer]
Did you use shouldDirty option for setValue. You can refer to the doc.

**Reply by MoSheikh** [reply]
Thank you for referring me back to that. I had previously tried with that option but I guess I had something else configured incorrectly at that point too. Using the option ```shouldDirty``` option now does exactly what I needed. 

Really appreciate your response and as well as the wonderful library!

**Reply by bluebill1049** [reply]
no worries.
